File: adhoc/118_pascals_triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle = []

        for i in range(numRows):
            if i == 0:
                triangle.append([1])
            elif i == 1:
                triangle.append([1,1])
            else:
                temp = [1]
                for j in range(i - 1):
                    temp.append(triangle[i-1][j] + triangle[i-1][j+1])
                temp.append(1)
                triangle.append(temp)
        return triangle
        

File: array/118_pascals_triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<List<Integer>>();
        res.add(Arrays.asList(1));
        if (numRows == 1) {
            return res;
        }
        
        for (int i = 1; i < numRows; i++) {
            List<Integer> prev = res.get(res.size() - 1);
            int prevSize = prev.size();
            List<Integer> temp = new ArrayList<>();
            temp.add(1);
            for (int j = 0; j < prevSize - 1; j++) {
                temp.add(prev.get(j) + prev.get(j+1));
            }
            temp.add(1);
            res.add(temp);
        }
        return res;
    }
    
}